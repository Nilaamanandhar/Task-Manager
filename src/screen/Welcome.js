import React, { Component } from 'react'
import { View, Text, ImageBackground, Button, StyleSheet } from 'react-native';
import welcomeImage from '../assests/image/welcome.jpg'
export default class Welcome extends Component {
      render() {
            return (
                  <ImageBackground source={welcomeImage} style={styles.welcomeContainer}>
                        <View style={styles.welcomeTextContainer}>
                              <Text style={styles.welcomeText}>WELCOME TO TODO NOTE</Text>
                              <Button title="Login" onPress={() => this.props.navigation.navigate('login')} />
                        </View>
                  </ImageBackground>
            )
      }
}

const styles = StyleSheet.create({
      welcomeContainer: {
            flex: 1,
            width: null,
            height: null,
            justifyContent: "space-between"
      },
      welcomeTextContainer: {
            display: "flex",
            flexDirection: "column",
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center',
            backgroundColor: 'rgba(0,128,128,0.7)'
      },
      welcomeText: {
            color: 'white',
            fontWeight: 'bold',
            marginBottom: 20,
            fontSize: 20,
            textAlign: "center"
      }

})
