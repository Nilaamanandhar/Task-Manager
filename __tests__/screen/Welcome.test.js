import 'react-native';
import React from 'react';

import Welcome from '../../src/screen/Welcome';

import renderer from 'react-test-renderer';

describe('<Welcome />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Welcome />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});