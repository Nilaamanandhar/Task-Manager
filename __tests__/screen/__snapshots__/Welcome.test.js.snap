// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome /> renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "height": null,
      "justifyContent": "space-between",
      "width": null,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assests/image/welcome.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": null,
          "width": null,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,128,128,0.7)",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      WELCOME TO TODO NOTE
    </Text>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
